### PR TITLE
Add UI to enable/disable experiments (feature flags)

### LINKED ISSUE #1738: Add Feature Flags UI
## Feature Description

Add a simple UI ( checkboxes ) to enable / disable feature flagged features. 

Follow up ticket from #1464

## Additional Context

Screenshot of Chrome's experiments screen to give an idea of what this could look like:

Screenshot 2020-05-12 at 15 07 54

Gutenberg experiments page:

https://github.com/WordPress/gutenberg/blob/master/lib/experiments-page.php
https://github.com/WordPress/gutenberg/blob/master/packages/e2e-tests/experimental-features.js

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

* Add new `WEBSTORIES_DEV_DEBUG` constant or similar. Defaults to `true`, but is `false` for stable releases.  
  Background: we want this to be only enabled on development builds, not stable releases.
* Enhance build tool to set this constant automatically
* Add new `Experiments` class that defines all feature flags, and — if `WEBSTORIES_DEV_DEBUG` is `true` — adds a new settings screen to the admin menu (using the Settings API).
* The way flags are defined will slightly changed, as they will now need to be grouped by area, and the titles will likely need to be translatable (as they will be displayed in the UI).
* In the `Settings` class, a new experiments/flags setting will be registered, which can be used on the settings screen

### PR DESCRIPTION
**Pull request by swissspidy**
## Relevant Technical Choices

* Experiments page is only visible for admins
* Once a flag has been turned on by default, there's no longer a checkbox for it in the UI.
* Experiments page will not be visible for the stable releases we ship, at least for now.

## To-do

Possible follow-up tickets:

* E2E tests for experimental features
* Remove flags that are enabled by default

## User-facing changes

New settings page under Stories -> Experiments:

Screenshot 2020-07-27 at 17 14 22

(visible only for admins)

## Testing Instructions

---

Fixes #1738

### CODE REVIEW

File: web-stories.php [hunk 1]
```diff
@@ -47,6 +47,7 @@
 define( 'WEBSTORIES_PLUGIN_DIR_URL', plugin_dir_url( WEBSTORIES_PLUGIN_FILE ) );
 define( 'WEBSTORIES_ASSETS_URL', WEBSTORIES_PLUGIN_DIR_URL . 'assets' );
 define( 'WEBSTORIES_MINIMUM_PHP_VERSION', '5.6.0' );
+define( 'WEBSTORIES_DEV_MODE', true );
```
Comment (spacedmonkey) [inline web-stories.php]:
```suggestion
if( ! defined( 'WEBSTORIES_DEV_MODE' ) ) {
    define( 'WEBSTORIES_DEV_MODE', true );
}
```

This is an example where site owner might want to turn this on or off.

Comment (spacedmonkey) [reply web-stories.php]: Should this not default to off / false?
We could set it to true here https://github.com/google/web-stories-wp/blob/main/bin/local-env/install-wordpress.sh#L131-L135

Comment (swissspidy) [reply web-stories.php]: The idea is to turn this constant **off** for any production builds (still needs a ticket).

_For now_ I don't think this should be changeable by users as the screen is currently only useful for the development team. It's pretty useless for anyone else.

I like the idea originally laid out in #1738, but we are simply not there yet.

Why? Most flags are for things that ar not even implemented yet. For example, the "Keyboard Shortcuts" flag displays a button that does nothing. Right now, there's no point in giving end users a way to toggle this flag.

This is a somewhat unanswered question, see https://github.com/google/web-stories-wp/issues/1738#issuecomment-627400233

Comment (spacedmonkey) [reply web-stories.php]: I still think that the suggested change is valid.

Comment (swissspidy) [reply web-stories.php]: Is that the only thing blocking you from approving otherwise?

File: tests/e2e/utils/experimentalFeatures.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import { visitAdminPage } from '@wordpress/e2e-test-utils';
+
+/**
+ * Toggle experiments state.
+ *
+ * @param {Array<string>}features Array of experiments to enable.
+ * @param {boolean} enable Whether the features should be enabled or not.
+ * @return {Promise<void>}
+ */
+async function toggleExperiments(features, enable) {
```
Comment (swissspidy) [inline tests/e2e/utils/experimentalFeatures.js:29]: FWIW, this util allows us to write tests for things that are still behind flags.

File: web-stories.php (hunk 1 above)
Comment (spacedmonkey) [reply web-stories.php]: It is the last thing and I will approve.

**Review by spacedmonkey — APPROVED** [review]
Looks good to me

**Comment by spacedmonkey** [comment]
@swissspidy I have this tweak related - https://github.com/google/web-stories-wp/pull/3715

**Comment by diegovar** [comment]
Some flags like `media3pTab` seem to have disappeared. Was that an unintentional change?

**Comment by diegovar** [comment]
Ah looks like showTextAndShapesSearchInput was added twice, and one of them was supposed to be media3pTab. Perhaps there should be a test to ensure that feature flag names are unique?

**Comment by diegovar** [comment]
I re-added `media3pTab` via #3620

**Comment by diegovar** [comment]
I don't seem to see the experiments option in the Stories tab, even though I'm an admin. Do I have to do anything else to have it appear?

**Comment by swissspidy** [comment]
@diegovar Heya, sorry for the inconvenience! Good point with the tests, looks like a copy/paste issue or merge issue. I'll send an email about this new feature soon, explaining how it works, etc.

**Pull request merged**